### PR TITLE
patchkernel: allow to pass a generic container to the functions that delete cells/vertices/interfaces

### PR DESCRIPTION
**Pull request by andrea-iob**
This avoid creating a list of vertices to delete during the adaption of the VolOctree patch.

### CODE REVIEW

File: src/patchkernel/patch_kernel.tpp [hunk 1]
```diff
@@ -45,6 +45,150 @@ std::unique_ptr<patch_t> PatchKernel::clone(const patch_t *original)
...
+		return false;
+	}
+
+	// Deleting the last interface requires some additional work. If the ids
+	// of interfaces to be deleted contain the last interface, that interface
+	// is deleted after deleting all other interfaces. In this way we make
+	// sure to delete the last interface just once (after deleting the last
+	// interface, another interface becomes the last one and that interface
+	// may be on the deletion list as well, and on and so forth).
+	std::size_t lastId;
```
Comment (DMurataj01) [inline src/patchkernel/patch_kernel.tpp:168]: think this lastId should be of type `long` / `id_t`

Comment (andrea-iob) [reply src/patchkernel/patch_kernel.tpp:168]: Will fix it, thanks.

**Review by marcocisternino — COMMENTED** [review]
Tested with dual solver porous tanks. It preserves the results of the master.
I could approve but I saw a comment asking for modification. I will approve when the modification is introduced.

File: src/patchkernel/patch_kernel.tpp (hunk 1 above)
Comment (andrea-iob) [reply src/patchkernel/patch_kernel.tpp]: Fixed.

**Review by marcocisternino — APPROVED** [review]

**Pull request merged**